metricbeat/module/rabbitmq/queue/data.go

```diff
@@ -45,7 +45,7 @@ var (
 		"consumers": s.Object{
 			"count": c.Int("consumers"),
```

Comment:
What value does `consumers` have when `consumer_utilisation` is null? It could be good to add a test case for this.

Reply:
```

Reply:
This is what we read from RabbitMQ
```
...
    "consumer_utilisation": null,
    "consumers": 0,
...
```

This is metricbeat's output for this case (notice empty `utilisation` element
```
  "rabbitmq": {
    "node": {
      "name": "rabbit@Pablos-MacBook-Pro"
    },
    "queue": {
      "state": "running",
      "name": "hello",
      "memory": {
        "bytes": 13384
      },
      "consumers": {
        "count": 0,
        "utilisation": {}
      },
      "auto_delete": false,
      "arguments": {},
      "messages": {
        "total": {
          "details": {
            "rate": 0
          },
          "count": 0
        },
        "ready": {
          "details": {
            "rate": 0
          },
          "count": 0
        },
        "unacknowledged": {
          "count": 0,
          "details": {
            "rate": 0
          }
        },
        "persistent": {
          "count": 0
        }
      }
```

it would be nice to have a test for this case, but I'm not sure how to add it given that current test structure only allows one mocked response per metrics endpoint URL
https://github.com/elastic/beats/blob/master/metricbeat/module/rabbitmq/mtest/server.go#L56

can you shed some light on it?

Reply:
Ok, leave it as is and we will add a test after https://github.com/elastic/beats/pull/10667

Reply:
tested temporarily using `null` for consumer utilisation. Tests succeed
```
diff --git a/metricbeat/module/rabbitmq/_meta/testdata/queue_sample_response.json b/metricbeat/module/rabbitmq/_meta/testdata/queue_sample_response.json
index 7c3088e6d..fd7b60b93 100644
--- a/metricbeat/module/rabbitmq/_meta/testdata/queue_sample_response.json
+++ b/metricbeat/module/rabbitmq/_meta/testdata/queue_sample_response.json
@@ -76,7 +76,7 @@
             "rate": 0.5
         },
         "idle_since": "2017-07-28 23:45:52",
-        "consumer_utilisation": 0.7,
+        "consumer_utilisation": null,
         "policy": null,
         "exclusive_consumer_tag": null,
         "consumers": 3,
diff --git a/metricbeat/module/rabbitmq/queue/queue_test.go b/metricbeat/module/rabbitmq/queue/queue_test.go
index 2f3320abb..a033a0350 100644
--- a/metricbeat/module/rabbitmq/queue/queue_test.go
+++ b/metricbeat/module/rabbitmq/queue/queue_test.go
@@ -54,8 +54,8 @@ func TestFetchEventContents(t *testing.T) {
        consumers := event["consumers"].(common.MapStr)
        utilisation := consumers["utilisation"].(common.MapStr)
        assert.EqualValues(t, 3, consumers["count"])
-       assert.EqualValues(t, 0.7, utilisation["pct"])
-
+       //(assert.EqualValues(t, 0.7, utilisation["pct"])
+       assert.Nil(t, utilisation["pct"])
        memory := event["memory"].(common.MapStr)
        assert.EqualValues(t, 232720, memory["bytes"])
```